From 0531dc3066e61e0634fe614c1418ba12de73e22c Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Wed, 23 Oct 2013 19:08:38 -0400
Subject: [PATCH 23/29] Add xmpp error check to _check_for_notification()

It looks like xmpp msg()'s can result in more than one clause
returned by the host, and that only one is processed per msg()
call. If so, it is possible that a failure may not be processed
until the notification call, which does not check.

Add a failure check to the notification method.
---
 cloudprint/xmpp.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/cloudprint/xmpp.py b/cloudprint/xmpp.py
index 3f0e0e3..b49b668 100644
--- a/cloudprint/xmpp.py
+++ b/cloudprint/xmpp.py
@@ -94,7 +94,14 @@ class XmppConnection(object):
 
     def _check_for_notification(self):
         """Check for any notifications which have already been received"""
-        return self._handler.get_elem() is not None
+        elem = self._handler.get_elem()
+
+        if elem:
+            assert not elem.tag.endswith('failure') and not elem.tag.endswith('error')
+            return True
+        else:
+            return False
+
 
     def _send_keepalive(self):
         LOGGER.info("Sending XMPP keepalive")
-- 
1.8.4.4

